buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/benchmark/BenchmarkingPlugin.kt

```diff
@@ -0,0 +1,260 @@
+
+private val NamedDomainObjectContainer<KotlinSourceSet>.commonMain
+    get() = maybeCreate("commonMain")
+
+private val NamedDomainObjectContainer<KotlinSourceSet>.nativeMain
+    get() = maybeCreate("nativeMain")
+
+private val NamedDomainObjectContainer<KotlinSourceSet>.jvmMain
+    get() = maybeCreate("jvmMain")
+
+private val Project.benchmark: BenchmarkExtension
+    get() = extensions.getByName(BenchmarkingPlugin.BENCHMARK_EXTENSION_NAME) as BenchmarkExtension
+
+private val Project.nativeWarmup: Int
+    get() = (property("nativeWarmup") as String).toInt()
+
+private val Project.jvmWarmup: Int
+    get() = (property("jvmWarmup") as String).toInt()
+
+private val Project.attempts: Int
+    get() = (property("attempts") as String).toInt()
+
+private val Project.nativeBenchResults: String
+    get() = property("nativeBenchResults") as String
+
+private val Project.jvmBenchResults: String
+    get() = property("jvmBenchResults") as String
+
+private val Project.compilerArgs: List<String>
+    get() = (findProperty("compilerArgs") as String?)?.split("\\s").orEmpty()
+
+internal val Project.kotlinVersion: String
+    get() = property("kotlinVersion") as String
+
+internal val Project.konanVersion: String
+    get() = property("konanVersion") as String
+
+internal val Project.kotlinStdlibVersion: String
+    get() = property("kotlinStdlibVersion") as String
+
+internal val Project.kotlinStdlibRepo: String
+    get() = property("kotlinStdlibRepo") as String
+
+internal val Project.nativeJson: String
+    get() = project.property("nativeJson") as String
+
+internal val Project.commonBenchmarkProperties: Map<String, Any>
+    get() = mapOf(
+        "cpu" to System.getProperty("os.arch"),
+        "os" to System.getProperty("os.name"),
+        "jdkVersion" to System.getProperty("java.version"),
+        "jdkVendor" to System.getProperty("java.vendor"),
+        "kotlinVersion" to kotlinVersion
+    )
+
+open class BenchmarkExtension @Inject constructor(val project: Project) {
+    var applicationName: String = project.name
+    var commonSrcDirs: Collection<Any> = emptyList()
+    var jvmSrcDirs: Collection<Any> = emptyList()
+    var nativeSrcDirs: Collection<Any> = emptyList()
+    var linkerOpts: Collection<String> = emptyList()
+}
+
+/**
+ * A plugin configuring a benchmark Kotlin/Native project.
+ */
+open class BenchmarkingPlugin: Plugin<Project> {
+
+    private val mingwPath: String = System.getenv("MINGW64_DIR") ?: "c:/msys64/mingw64"
+
+    private fun Project.determinePreset(): KotlinNativeTargetPreset =
+        defaultHostPreset(this).also { preset ->
+            logger.quiet("$project has been configured for ${preset.name} platform.")
+        } as KotlinNativeTargetPreset
+
+    private fun Project.configureSourceSets(kotlinVersion: String) {
+        with(kotlin.sourceSets) {
+            commonMain.dependencies {
+                implementation("org.jetbrains.kotlin:kotlin-stdlib-common:$kotlinStdlibVersion")
```

Comment:
Can there be situations when stdlib version isn't equal kotlin version? We use it from kotlin and kotlin-native? If we will separate it, then I should add it to report, because difference in stdlib can influence on results too.

Reply:
Usually Kotlin compiler and stdlib versions are equal but in general case they can be changed independently (see the root `gradle.properties` - we have separate properties for these versions there). Kotlin stdlib version may influence our performance because we use the same common part in our stdlib.

cc @vvlevchenko